chain/internal/rescan/rescan.go

```diff
@@ -0,0 +1,20 @@
+package rescan
+
+import "github.com/lightninglabs/neutrino"
+
+type Updater interface {
```

Comment:
Nice to extract those out into an interface. Though I don't think we need an extra sub package. If we want to make them internal, we just don't export them (so `rescanUpdater`, `rescanStarter`, `rescan` as names).

Also, all interfaces should get Godoc comments to explain what they represent.